lib/docker/imageservice.go

```diff
@@ -336,29 +365,75 @@ func initTransport(req RegistryConnectionRequest) (*http.Transport, string) {
 		DisableKeepAlives:   true,
 	}
 
+	// Figure out the registry address scheme (http or https). If the scheme
+	// was specified explicitly, keep it as-is. Otherwise, default to https
+	// unless the insecure flag was provided.
+	registryAddress, err := url.Parse(req.RegistryAddress)
```

Comment:
`url.Parse` will fail on addresses like `<ip>:5000` without a scheme.

Reply:
Good catch, this is actually why CI is failing.